### PR TITLE
add create usr_data

### PR DESCRIPTION
**Pull request by zhang-shengping**

### CODE REVIEW

File: app/waf/src/controllers/adc.controller.ts [hunk 1]
```diff
@@ -346,4 +349,24 @@ export class AdcController {
       }
     });
   }
+
+  private async cUserdata(): Promise<string> {
```
Comment (zhaoqin-github) [inline app/waf/src/controllers/adc.controller.ts]: What is cUserdata ?

File: app/waf/src/controllers/adc.controller.ts [hunk 2]
```diff
@@ -346,4 +349,24 @@ export class AdcController {
...
+    let root_password: string = Math.random()
+      .toString(36)
+      .slice(-8);
+    let admin_password: string = Math.random()
+      .toString(36)
+      .slice(-8);
+
+    const user_data_unencoded: string = `#cloud-config
+    runcmd:
+     - "echo"root:{{ ${root_password} }}" | chpasswd"
```
Comment (zongzw) [inline app/waf/src/controllers/adc.controller.ts]: remove {{}}, that's postman's variable definition.

Comment (zhang-shengping) [reply app/waf/src/controllers/adc.controller.ts]: roger

File: app/waf/src/controllers/adc.controller.ts [hunk 3]
```diff
@@ -344,4 +347,24 @@ export class AdcController {
...
+    let root_password: string = Math.random()
+      .toString(36)
+      .slice(-8);
+    let admin_password: string = Math.random()
+      .toString(36)
+      .slice(-8);
+
+    const user_data_unencoded: string = `#cloud-config
+    runcmd:
+     - "echo \\"root:${root_password}\\" | chpasswd"
```
Comment (zongzw) [inline app/waf/src/controllers/adc.controller.ts]: From the second line on, the format is incorrect, there should be no indents. 
Just want to suggest you have the test well done before you request the review.

File: app/waf/src/controllers/adc.controller.ts [hunk 4]
```diff
@@ -344,4 +347,24 @@ export class AdcController {
...
   }
+
+  private async cUserdata(): Promise<string> {
+    let root_password: string = Math.random()
+      .toString(36)
+      .slice(-8);
+    let admin_password: string = Math.random()
+      .toString(36)
+      .slice(-8);
+
```
Comment (Niklaus-xie) [inline app/waf/src/controllers/adc.controller.ts]: user_data content looks straightforward. although I doubt we really want a random password.

Comment (zhang-shengping) [reply app/waf/src/controllers/adc.controller.ts]: I can change it if all of you are happy to change.

**Comment by zhang-shengping** [comment]
TEST
```
USERDATA is as blew:

[root@63d9f5c066a6 wdir]# echo I2Nsb3VkLWNvbmZpZwogICAgcnVuY21kOgogICAgIC0gImVjaG8gXCJyb290Oml6dTE3dm45XCIgfCBjaHBhc3N3ZCIKICAgICAtICJlY2hvIFwiYWRtaW46ZmVoejBzMG5cIiB8IGNocGFzc3dkIg== | base64 -d
#cloud-config
    runcmd:
     - "echo \"root:izu17vn9\" | chpasswd"
     - "echo \"admin:fehz0s0n\" | chpasswd"

TEST with CURL:

[root@63d9f5c066a6 wdir]# curl -k -g -i -X POST https://xxxxxxxf5net.com:8774/v2.1/6dd232a3f8db49b98864442210cf019f/servers -H "Accept: application/json" -H "Content-Type: application/json" -H "User-Agent: python-novaclient" -H "X-Auth-Token: $TOKEN" -d '{"server": {"name": "vm_userdata", "imageRef": "98e606f0-51dc-4a75-aba2-ea96071fcb1b", "flavorRef": "93a9c137-cf4a-42f6-9881-97e38ae30082", "user_data": "I2Nsb3VkLWNvbmZpZwogICAgcnVuY21kOgogICAgIC0gImVjaG8gXCJyb290Oml6dTE3dm45XCIgfCBjaHBhc3N3ZCIKICAgICAtICJlY2hvIFwiYWRtaW46ZmVoejBzMG5cIiB8IGNocGFzc3dkIg==", "max_count": 1, "min_count": 1, "networks": [{"uuid": "2bfdaa3a-b3c7-414a-934d-1d47751bef4d"}], "config_drive": true}}'
HTTP/1.1 202 Accepted
Date: Thu, 18 Apr 2019 11:08:03 GMT
Server: Apache/2.4.18 (Ubuntu)
Content-Length: 466
Location: https://xxxxxxx.f5net.com:8774/v2.1/6dd232a3f8db49b98864442210cf019f/servers/19d2d39b-98c0-4a39-b198-b1531456c02d
Content-Type: application/json
Openstack-Api-Version: compute 2.1
X-Openstack-Nova-Api-Version: 2.1
Vary: OpenStack-API-Version,X-OpenStack-Nova-API-Version
X-Openstack-Request-Id: req-75dbd56e-93e4-4dde-8108-cb83159b2905
X-Compute-Request-Id: req-75dbd56e-93e4-4dde-8108-cb83159b2905

{"server": {"security_groups": [{"name": "default"}], "OS-DCF:diskConfig": "MANUAL", "id": "19d2d39b-98c0-4a39-b198-b1531456c02d", "links": [{"href": "https://xxxxxx.f5net.com:8774/v2.1/6dd232a3f8db49b98864442210cf019f/servers/19d2d39b-98c0-4a39-b198-b1531456c02d", "rel": "self"}, {"href": "https://xxxxxxx.f5net.com:8774/6dd232a3f8db49b98864442210cf019f/servers/19d2d39b-98c0-4a39-b198-b1531456c02d", "rel": "bookmark"}], "adminPass": "hzUZbveMH9BN"}}

Log on:
➜  ~ ssh root@10.145.64.34
Password:
[root@localhost:NO LICENSE:Standalone] config #
```

File: app/waf/src/controllers/adc.controller.ts (hunk 3 above)
Comment (zhang-shengping) [reply app/waf/src/controllers/adc.controller.ts]: At jikui second approval, it is testd with GUI first, then it is tested with `curl` CLI

**Pull request merged**